### PR TITLE
chore: add more metrics for perf

### PR DESCRIPTION
**Pull request by will-2012**
### Description

Add more metrics for perf.

### Rationale

N/A.

### Example

`--metrics.expensive` enable more metrics for perf.

### Changes

Notable changes:
* N/A.

### CODE REVIEW

File: core/blockchain.go [hunk 1]
```diff
@@ -1882,12 +1882,10 @@ func (bc *BlockChain) insertChain(chain types.Blocks, setHead bool) (int, error)
...
 		accountHashTimer.Update(statedb.AccountHashes)                  // Account hashes are complete(in validation)
 		storageHashTimer.Update(statedb.StorageHashes)                  // Storage hashes are complete(in validation)
-		triehash := statedb.AccountHashes + statedb.StorageHashes       // The time spent on tries hashing
-		trieUpdate := statedb.AccountUpdates + statedb.StorageUpdates   // The time spent on tries update
 		trieRead := statedb.SnapshotAccountReads + statedb.AccountReads // The time spent on account read
 		trieRead += statedb.SnapshotStorageReads + statedb.StorageReads // The time spent on storage read
-		blockExecutionTimer.Update(ptime - trieRead)                    // The time spent on EVM processing
-		blockValidationTimer.Update(vtime - (triehash + trieUpdate))    // The time spent on block validation
+		blockExecutionTimer.Update(ptime)                               // The time spent on EVM processing
+		blockValidationTimer.Update(vtime)                              // The time spent on block validation
```
Comment (bendanzhentan) [inline core/blockchain.go:1888]: It seems that the definitions of `blockExecutionTimer` and `blockValidationTimer` have been re-defined. While I am not sure if this is intended. if yes, could you please explain why this change is better?

File: core/rawdb/accessors_snapshot.go [hunk 2]
```diff
@@ -74,6 +76,10 @@ func DeleteSnapshotRoot(db ethdb.KeyValueWriter) {
 
 // ReadAccountSnapshot retrieves the snapshot entry of an account trie leaf.
 func ReadAccountSnapshot(db ethdb.KeyValueReader, hash common.Hash) []byte {
+	start := time.Now()
+	if metrics.EnabledExpensive {
+		defer func() { rawdbGetAccountSnapNodeTimer.UpdateSince(start) }()
+	}
```
Comment (bendanzhentan) [inline core/rawdb/accessors_snapshot.go:82]:
```suggestion
	if metrics.EnabledExpensive {
		start := time.Now()
		defer func() { rawdbGetAccountSnapNodeTimer.UpdateSince(start) }()
	}
```

File: core/state_processor.go [hunk 3]
```diff
@@ -99,6 +106,9 @@ func (p *StateProcessor) Process(block *types.Block, statedb *state.StateDB, cfg
 		}
 		receipts = append(receipts, receipt)
 		allLogs = append(allLogs, receipt.Logs...)
+		if metrics.EnabledExpensive {
+			processTxTimer.UpdateSince(start)
+		}
```
Comment (bendanzhentan) [inline core/state_processor.go:111]: IMO we should call `processTxTimer.UpdateSince(start)` even if `applyTransaction()` fails.

File: core/rawdb/accessors_snapshot.go (hunk 2 above)
Comment (will-2012) [reply core/rawdb/accessors_snapshot.go]: Updated.

File: core/state_processor.go (hunk 3 above)
Comment (will-2012) [reply core/state_processor.go:118]: Fixed.

File: core/blockchain.go (hunk 1 above)
Comment (will-2012) [reply core/blockchain.go:1888]: Yes; this PR does not delete any metrics, but only adjusts the calculation method of metrics, because when analyzing performance bottlenecks during `--metrics.expensive`, blockExecutionTimer/blockValidationTimer/blockWriteTimer may be negative when deducting concurrency latency. This is a bit confusing. In fact, if we pay attention to the metrics of internal concurrent actions, we can directly configure a corresponding panel, and  I think it is better to have clear semantics of blockExecutionTimer/blockValidationTimer/blockWriteTimer.   :)
This is just a slight change and ultimately depends on the reviewers' suggestions.

Comment (will-2012) [reply core/blockchain.go:1888]: Without `--metrics.expensive`, These `triehash/trieUpdate/..Commits` are all equal to 0.

**Review by bendanzhentan — APPROVED** [review]
LGTM

File: core/blockchain.go (hunk 1 above)
Comment (bendanzhentan) [reply core/blockchain.go:1888]: ACK. Thanks.

**Comment by owen-reorg** [comment]
Close this PR since most of the metrics have been added in https://github.com/bnb-chain/op-geth/pull/71
